src/components/McuUpdateDialogView.tsx

```diff
@@ -241,121 +238,122 @@ Are you sure you want to continue?`,
-            )}
-            {!writingHasStarted && showDelayTimeout && (
-                <Form.Group className="upload-delay p-3">
-                    <div className="d-flex justify-content-between">
-                        <Form.Label className="mb-0">
-                            <strong>
-                                Keep default delay after image upload
-                            </strong>
-                            <OverlayTrigger
-                                placement="bottom-end"
-                                overlay={
-                                    <Tooltip id="tooltip-delay-info">
-                                        <div className="info text-left">
-                                            <span>{TOOLTIP_TEXT}</span>
-                                        </div>
-                                    </Tooltip>
-                                }
-                            >
-                                <span className="mdi mdi-information-outline info-icon ml-1" />
-                            </OverlayTrigger>
-                        </Form.Label>
-                        <Toggle
-                            onToggle={toggleDefaultTimeoutUI}
-                            isToggled={keepDefaultTimeout}
+                </div>
+                {writing && (
+                    <div className="tw-flex tw-flex-col tw-gap-2">
+                        <div>
+                            <strong>Status: </strong>
+                            <span>{` ${
+                                progress ? progress.message : 'Starting...'
+                            }`}</span>
+                        </div>
+                        <ProgressBar
+                            hidden={!writing}
+                            now={progress?.stepProgressPercentage ?? 0}
+                            style={{ height: '4px' }}
                         />
                     </div>
-                    <FormLabel
-                        className={classNames(
-                            'w-100 mb-0 mt-3',
-                            keepDefaultTimeout ? 'd-none' : 'd-block'
-                        )}
-                    >
-                        <div className="d-flex justify-content-between mb-2 flex-row">
-                            <div>
-                                <strong>Set delay duration</strong>
-                            </div>
-                            <div className="d-flex">
-                                <NumberInlineInput
-                                    value={uploadDelay}
-                                    range={uploadDelayRange}
-                                    onChange={updateUploadDelayTimeout}
-                                />
-                                <span>sec</span>
+                )}
+                {!writingHasStarted && showDelayTimeout && (
+                    <div className="tw-flex tw-flex-col tw-gap-4 tw-border tw-border-solid tw-border-gray-100 tw-p-3">
```

Comment:
it was p-3 padding from bootstrap, does it match the one from tailwind? If yes ignore this comment :)